Makefile

```diff
@@ -1,32 +1,144 @@
+# OpenEdx E-Commerce
 EDX_EC_ARCHIVE_URL ?= https://github.com/edx/ecommerce/archive/master.tar.gz
+EDX_EC_RELEASE_REF ?= master
+EDX_EC_DOCKER_TAG  ?= latest
+
+# Get local user ids
+DOCKER_UID = $(shell id -u)
+DOCKER_GID = $(shell id -g)
+
+# Docker
+COMPOSE          = \
+  DOCKER_UID=$(DOCKER_UID) \
+  DOCKER_GID=$(DOCKER_GID) \
+  EDX_EC_ARCHIVE_URL=$(EDX_EC_ARCHIVE_URL) \
+  EDX_EC_RELEASE_REF=$(EDX_EC_RELEASE_REF) \
+  EDX_EC_DOCKER_TAG=$(EDX_EC_DOCKER_TAG) \
+  docker-compose
+COMPOSE_RUN      = $(COMPOSE) run --rm -e HOME="/tmp"
+COMPOSE_EXEC     = $(COMPOSE) exec
+MANAGE           = $(COMPOSE_RUN) app python manage.py
+
+# Terminal colors
+COLOR_DEFAULT = \033[0;39m
+COLOR_ERROR   = \033[0;31m
+COLOR_INFO    = \033[0;36m
+COLOR_RESET   = \033[0m
+COLOR_SUCCESS = \033[0;32m
+COLOR_WARNING = \033[0;33m
+
+# Shell
+SHELL=bash
+
 
 default: help
 
+data/assets/.keep:
+	mkdir -p data/assets
+	touch data/assets/.keep
+
+data/media/.keep:
+	mkdir -p data/media
+	touch data/media/.keep
+
 src/.keep:
 	mkdir -p src
 	touch src/.keep
 
 src/README.rst:
-	curl -Lo /tmp/ecommerce.tgz $(EDX_EC_ARCHIVE_URL)
+	@echo -e "$(COLOR_INFO)Downloading sources archive...$(COLOR_RESET)"
+	curl -sLo /tmp/ecommerce.tgz $(EDX_EC_ARCHIVE_URL)
 	tar xzf /tmp/ecommerce.tgz -C ./src --strip-components=1
 
+bootstrap: \
+  clean \
+  build \
+  dev-assets \
+  migrate
+bootstrap:  ## Bootstrap the application
+	@echo -e "$(COLOR_SUCCESS)Project bootstrapped successfully.$(COLOR_RESET)"
+	@echo -e "Now try to run the development server using: $(COLOR_INFO)make dev$(COLOR_RESET)"
+.PHONY: bootstrap
+
 build: \
+  info \
   tree \
   fetch-src
 build:  ## Build docker image
-	docker build -t edxec:latest .
+	@echo -e "$(COLOR_INFO)Starting Docker image build...$(COLOR_RESET)"
+	$(COMPOSE) build app
 .PHONY: build
 
-clean:  ## Remove downloaded sources
+clean:  ## Remove downloaded sources, assets and database
+	@echo -e "$(COLOR_WARNING)Removing downloaded sources...$(COLOR_RESET)"
 	rm -fr src
+	@echo -e "$(COLOR_WARNING)Removing collected assets...$(COLOR_RESET)"
+	rm -fr data
+	@echo -e "$(COLOR_WARNING)Removing database...$(COLOR_RESET)"
+	${MAKE} stop
+	$(COMPOSE) rm mysql
```

Comment:
maybe use `-f` to don't ask to remove the container ?

Reply:
Or if you want the approval for removing it, maybe you should first delete the container and then remove the sources and assets ?

Reply:
I prefer letting the user make its own choice and not removing the database by default, so I opted for the second.